pkg/app/piped/executor/lambda/lambda.go

```diff
@@ -90,7 +90,20 @@ func apply(ctx context.Context, in *executor.Input, cloudProviderName string, cl
 		return false
 	}
 
-	if err := client.Apply(ctx, fm, cloudProviderCfg.Role); err != nil {
+	ok, err := client.AvailableFunctionName(ctx, fm.Spec.Name)
+	if err != nil {
+		in.LogPersister.Errorf("Unable to check avalability of given function name %s: %v", fm.Spec.Name, err)
+		return false
+	}
+	if !ok {
+		// Try to delete the function which name overlap with given function name in lambda function manifest.
+		if err := client.DeleteFunction(ctx, fm.Spec.Name); err != nil {
```

Comment:
Is it ok for running function/real-traffic if we delete it?

Reply:
Nice catch! 👍 The traffic will be lost on timeout handshake failed with Lambda app or behold on upstream, I think it depends. I understand your wonder, we should diversion the traffic in a more reasonable way instead of just drop it. 👍  Basically, we could implement this `DeleteFunction` in a more flexible way by using [Versioning Lambda function](https://docs.aws.amazon.com/lambda/latest/dg/configuration-versions.html).
In that case, the steps will be like
- Confirm FunctionName exists? ( if not, just create a new one using that name )
- Create a new version of Function attach with the given function name
- Promote traffic to the new version of Function
- Remove the old version of Function

That what I planned to do next once the deployment logic of the `LAMBDA_PROMOTE` stage is implemented.
( above step by step still in incubating, please feel free to comment for that part too 😄 )

Reply:
Do they provide any way to update the function? Update for the existing function.

Reply:
Yes, they're. But as far as I investigated, only functions which not be marked as published version can be updated.
ref: [Lambda.UpdateFunctionCode](https://docs.aws.amazon.com/sdk-for-go/api/service/lambda/#Lambda.UpdateFunctionCode)
ref: https://github.com/pipe-cd/pipe/pull/1432#issuecomment-760600051

Reply:
Currently, I'm investing around the versioning lambda function ( publish/unpublish.. ). If we have a way to make only one published version of a Lambda function available to be called at a time, then it may be a better option for `LAMBDA_PROMOTE` stage implementation than the above-suggested plan.